tests_backend/run_backend_tests.ts

```diff
@@ -189,69 +205,108 @@ async function testLoop(functionsUrl: URL, supabase: SupabaseType): Promise<bool
   return ok
 }
 
-async function runTests() {
-  if (!functionsUrl) {
-    p.log.error('No functions URL, cannot run tests')
-    Deno.exit(1)
-  }
-
-  if (!supabase) {
-    p.log.error('No supabase connection, cannot run tests')
-    Deno.exit(1)
-  }
-
-  let ok = await testLoop(functionsUrl, supabase)
-  // This is likely unreacheble
-  if (!ok)
-    Deno.exit(1)
-
+async function runTestsForFolder(folder: string, shortName: string, firstArg: string) {
   const useLocalRedis = Deno.env.get('USE_LOCAL_REDIS') === 'true'
-  if (useLocalRedis) {
-    p.log.info('Running with redis')
-    killSupabase()
-    p.log.info('Starting supabase backend with local redis...')
-    // Local means redis is running on localhost
-    await startSupabaseBackend('local')
-    ok = await testLoop(functionsUrl, supabase)
-    if (!ok)
-      Deno.exit(1)
-  }
-
-  else { p.log.warn('Skipping running with redis') }
-
   const upstashToken = Deno.env.get('UPSTASH_TOKEN')
   const upstashUrl = Deno.env.get('UPSTASH_URL')
+
   if (upstashToken && upstashUrl) {
-    p.log.info('Running with upstash')
-    killSupabase()
     p.log.info('Generating temp ENV file for upstash...')
     tempUpstashEnvFilePath = await genTempUpstashEnvFile(upstashToken, upstashUrl)
     p.log.info(`Temp ENV file generated at ${tempUpstashEnvFilePath}`)
-    p.log.info('Starting supabase backend with upstash...')
-    // upstash means redis is running remotly and is being hostend by upstash
-    await startSupabaseBackend('upstash', tempUpstashEnvFilePath)
-    ok = await testLoop(functionsUrl, supabase)
+  }
+
+  for await (const dirEntry of Deno.readDir(folder)) {
+    const firstArg = Deno.args[0]
+
+    // this last conditions checks if we are supposed to run all tests or just the specific one
+    // Like /test cli will run only tests from the cli folder
+    if (dirEntry.isDirectory || !dirEntry.name.endsWith('.ts') || (firstArg !== 'all' ? shortName !== firstArg : false))
```

Comment:
could we change it to 
```suggestion
    if (dirEntry.isDirectory || !dirEntry.name.endsWith('.ts') || (firstArg !== 'all' && shortName !== firstArg))
```

the result is the same but it seems more readable